### PR TITLE
Add feature to autogenerate the OpenAPI for the webhook from the mock webhook server

### LINKED ISSUE #51: Implement OpenAPI generation for webhook API
# Goal
The goal of this PR is to generate the OpenAPI spec for the webhook callback interface.

# Discussion
- `make mock-webhook` runs an `actix` web server in Rust. 
- `apistos` is a wrapper that can generate OpenApi specs from an `actix` implementation.
- This is a time-boxed activity to see if auto generation will work.

# Acceptance Criteria
- [ ] OpenApi spec file is generated and added to repo.
- [x] (Optional) OpenAPI interface is generated for testing and example.

### PR DESCRIPTION
**Pull request by mattheworris**
# Goal
The goal of this PR is to generate the OpenAPI spec for the webhook callback interface.

# Discussion
- `make mock-webhook` runs an `actix` web server in Rust. 
- `apistos` is a wrapper that can generate OpenApi specs from an `actix` implementation.
- This is a time-boxed activity to see if auto generation will work.

# Acceptance Criteria
- [x] OpenApi spec file is generated and added to repo.
- [x] (Optional) OpenAPI interface is generated for testing and example.

Closes #51

### CODE REVIEW

File: rust-webhook-server/src/api.rs [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+        handle: callback_msg.handle,
+    };
+    Ok(CreatedJson(payload))
+}
+
+#[api_operation(summary = "Health check")]
+pub(crate) async fn health_check() -> Result<Json<HealthResponse>, Error> {
+    println!("Health check");
+    Ok(Json(HealthResponse {
+        message: "Server is healthy".to_string(),
```
Comment (claireolmstead) [inline rust-webhook-server/src/api.rs:51]: why do you need to do to_String?

**Review by claireolmstead — APPROVED** [review]
LGTM

File: rust-webhook-server/src/api.rs (hunk 1 above)
Comment (mattheworris) [reply rust-webhook-server/src/api.rs:51]: "Server is healthy" is a `&str`, which is known as a string slice. `message` is type `String`, which is a different type.
Using a `&str` to create a `String` is a common pattern in rust.
Using the `String::from` method: `message: String::from("Server is healthy")` also works.

**Review by claireolmstead — APPROVED** [review]
Tests run:
Run when not connected => see timeout and exit
- worker ✅ 
- api ✅ 

Is connected, run, disconnect => see timeout and exit 
- worker ✅ 
- api ✅ 

Is connected, run worker, wait 30s, stay connected => see no exit
- worker ✅ 
- api ✅ 

🥇 🥇 🥇

**Pull request merged**